transport/http/src/test/java/org/kaazing/gateway/transport/http/security/auth/MismatchedAuthSchemeSendsExtended401AuthTestIT.java

```diff
@@ -0,0 +1,189 @@
+import org.kaazing.gateway.transport.IoHandlerAdapter;
+import org.kaazing.gateway.transport.http.HttpAcceptSession;
+import org.kaazing.gateway.transport.http.HttpAcceptorRule;
+import org.kaazing.gateway.transport.http.HttpStatus;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+import org.kaazing.test.util.MethodExecutionTrace;
+
+/*
+ * These tests are used to verify the behavior of the HttpSubjectSecurityFilter class
+ * against possible outcomes of login module chain authentication.
+ */
+public class MismatchedAuthSchemeSendsExtended401AuthTestIT {
+
+    private static final String REALM_NAME = "Test Realm";
+    private static final String BIND_ADDRESS = "http://localhost:8001/echoAuth";
+    private static final String APPLICATION_TOKEN_CHALLENGE_SCHEME = "Application Token";
+    private static final String[] ANY_ROLE = new String[] {"*"};
+    private static final String[] EMPTY_STRING_ARRAY = new String[]{};
+    private static final Object[] ADDITIONAL_CHALLENGES = new Object[]{"param=\"value\""};
+
+    private static final IoHandlerAdapter<HttpAcceptSession> HTTP_ACCEPT_HANDLER = new IoHandlerAdapter<HttpAcceptSession>() {
+        @Override
+        protected void doSessionOpened(HttpAcceptSession session) throws Exception {
+            session.setStatus(HttpStatus.SUCCESS_OK);
+            session.close(false);
+        }
+    };
+
+    private final HttpAcceptorRule acceptor = new HttpAcceptorRule();
+    private final TestRule trace = new MethodExecutionTrace();
+    private final K3poRule k3po = new K3poRule();
+    private final TestRule timeoutRule = new DisableOnDebug(new Timeout(5, SECONDS));
+
+    private LoginContextFactory loginContextFactoryMock;
+    private ResultAwareLoginContext loginContextMock;
+    private DefaultLoginResult loginResultMock;
+    private HttpRealmInfo[] realms;
+
+    @Rule
+    public JUnitRuleMockery context = new JUnitRuleMockery() {
+        {
+            setImposteriser(ClassImposteriser.INSTANCE);
+            setThreadingPolicy(new Synchroniser());
+        }
+    };
+
+    @Rule
+    public TestRule chain = RuleChain.outerRule(trace).around(acceptor).around(k3po).around(timeoutRule);
+
+    @Before
+    public void setUp() {
+        loginContextFactoryMock = context.mock(DefaultLoginContextFactory.class);
+        loginContextMock = context.mock(ResultAwareLoginContext.class);
+        loginResultMock = context.mock(DefaultLoginResult.class);
+
+        realms = new HttpRealmInfo[1];
+        realms[0] = new DefaultHttpRealmInfo(REALM_NAME, APPLICATION_TOKEN_CHALLENGE_SCHEME, REALM_NAME, EMPTY_STRING_ARRAY,
+                EMPTY_STRING_ARRAY, EMPTY_STRING_ARRAY, loginContextFactoryMock, null);
+
+        acceptor.getAcceptOptions().put("http.realms", realms);
+    }
+
+    @Test
+    @Specification("shouldChallengeDueToMismatchingAuthSchemes")
+    public void shouldChallengeDueToMismatchingAuthSchemes2()
```

Comment:
After merging with `shouldNotChallengeDueToMismatchingAuthSchemes2`, make sure to remove the 2 from the name.